src/main/java/com/unstoppabledomains/resolution/naming/service/uns/UNSInternal.java

```diff
@@ -139,9 +141,10 @@ public String getTokenUri(BigInteger tokenID) throws NamingServiceException {
   @Override
   public String getDomainName(BigInteger tokenID) throws NamingServiceException {
     try {
-      String registryAddress = this.getRegistryAddress(tokenID);
-      Registry registryContract = new Registry(blockchainProviderUrl, registryAddress, provider);
-      String domainName = registryContract.getDomainName(tokenID);
+      String tokenURI = this.getTokenUri(tokenID);
+      JsonProvider provider = new JsonProvider();
+      TokenUriMetadata metadata = provider.request(tokenURI, TokenUriMetadata.class);
+      String domainName = metadata.getName();
       if (domainName == null) {
```

Comment:
@JohnnyJumper Any validation for received domain name? Please read pivotal story description again.